c3/generator/devices.py

```diff
@@ -163,7 +166,7 @@ def __init__(self, **props):
         self.inputs = props.pop("inputs", 1)
         self.outputs = props.pop("outputs", 1)
 
-    def process(self, instr, chan, mixed_signal):
+    def process(self, instr: Instruction, chan: str, mixed_signal: dict) -> dict:
```

Comment:
A good practice for using `dict` in type annotations is to instead use the `Dict` from `Typing` which allows you to mention what the dictionary structure looks like. This has tripped me up in many places because we use `dict` everywhere in C3 and it is not always clear what my `dict` should look like in terms of the datatypes of the keys and values. I would recommend updating the dict annotations in this PR as much as possible to `Dict`